.github/workflows/deploy-core.yaml

```
@@ -0,0 +1,208 @@
+        run: |
+          pnpm vercel --token=${{ secrets.VERCEL_TOKEN }} env pull --environment=production .env
+
+      - name: Run Checks for Lint, Types, and Spell
+        run: pnpm ci:lint
+
+  test-non-prod:
+    if: ${{ github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/heads/release/') }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)